### ISSUE TITLE
Joystick PWM Option

### ISSUE THREAD

**Issue by Jonny0** [Ryochan7/antimicro#201]
I would like it if you could add a pulse width modulation option for the joystick button if you haven't already. Basically if the joystick was 50% to the left the left button whatever that maybe be would be turned on for 1ms then turned off for 1ms over and over or for 90% left on 9ms and off 1ms. Also if you could do that for the triggers that would be nice. Something like that would be very nice for racing games.
Thank You

**Comment by Ryochan7** [comment]
Something similar to this is possible. You would have to utilize turbo and use the Gradient mode for the relevant axis directions. Based on how much the joystick has been moved, the duration of key presses will increase while the duration of key releases will decrease. The amount of time that a key press and key release is experienced will also be dependent on the delay specified. At 100%, a key will be held.

![antimicro_gradient_20150617](https://cloud.githubusercontent.com/assets/581968/8222401/8004ea28-1530-11e5-8c0b-74677f2987b8.png)

For an example of how Gradient mode behaves, I had recorded some footage of FlatOut 2 being played using that mode to emulate rapid key presses and releases. The video is still available to view on Dailymotion.

www.dailymotion.com/video/x2edsrt_antimicro-gradient-test-flatout-2_videogames

Playing FlatOut 2 using its built-in controller support is the better experience but using Gradient mode is doable.
